### PR TITLE
71 adding jmespath and options to jsonformatting in testbox

### LINKED ISSUE #71: Adding JMESPath and options to JsonFormatting in testbox

### PR DESCRIPTION
**Pull request by Gormiee**
proto file changed to IsJsonQuary. JsonField the text.
should work with all json.
StringCleanUp cleans the string from some characters. might need to be changes.
Might still need some premadse options for string cleaning for the JsonField to look nicer.

### CODE REVIEW

File: src/UICrafter.Client/Components/Properties/AppViewProperties.razor [hunk 1]
```diff
@@ -110,6 +111,12 @@
     {
         if (AppView is null) return;
 
+        if (string.IsNullOrEmpty(AppView.Name))
```
Comment (Nephim) [inline src/UICrafter.Client/Components/Properties/AppViewProperties.razor]: IsNullOrWhitespace as a space should not be allowed as a name

File: src/UICrafter.Client/Components/Properties/ComponentProperties.razor [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    private void DeleteSelectedComponent()
+    {
+        if (SelectedComponent is null) return;
+
+        if (SelectedComponent.ComponentCase == UIComponent.ComponentOneofCase.Button)
+        {
+            var textBoxesWithSourceRef = UIComponents
+                .Where(component => component.ComponentCase == UIComponent.ComponentOneofCase.Textbox && component.Textbox.SourceRef == SelectedComponent.Guid)
+                .ToList();
```
Comment (Nephim) [inline src/UICrafter.Client/Components/Properties/ComponentProperties.razor]: Dont call ToList() it costs and is not needed when you just do a foreach after.

File: src/UICrafter.Core/Components/Render/RenderTextBox.razor [hunk 3]
```diff
@@ -27,9 +29,15 @@
     {
         if (string.IsNullOrEmpty(Textbox.Textbox.JsonField))
```
Comment (Aagaard00) [inline src/UICrafter.Core/Components/Render/RenderTextBox.razor:30]: Der er 2 forskellige felter til det, så en af dem vil altid være empty

File: src/UICrafter.Client/Components/Properties/Components/TextboxProperties.razor [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    [Parameter]
+    public required ObservableCollection<UIComponent> UIComponents { get; set; }
+
+    private void RadioChanged(bool isQuery)
+    {
+        SelectedComponent.Textbox.IsJsonQuery = isQuery;
+
+        (SelectedComponent.Textbox.JsonField) = (SelectedComponent.Textbox.JsonField);
+        Log.Debug(SelectedComponent.Textbox.JsonField + ", " + SelectedComponent.Textbox.IsJsonQuery);
```
Comment (Aagaard00) [inline src/UICrafter.Client/Components/Properties/Components/TextboxProperties.razor]: ValueChanged -> til en bind til RadioGroup

https://mudblazor.com/components/radio#dense

**Review by Aagaard00 — CHANGES_REQUESTED** [review]
Fixes plzz

File: src/UICrafter.Core/Components/Render/RenderTextBox.razor [hunk 5]
```diff
@@ -27,9 +29,15 @@
...
         {
-            Content = "No JsonField specified.";
+            Content = "No JsonField or JsonQuery specified.";
         }
-        else
+        else if (Textbox.Textbox.IsJsonQuery)
+        {
+            var _jmesPath = new JmesPath();
```
Comment (Aagaard00) [inline src/UICrafter.Core/Components/Render/RenderTextBox.razor]: Validering for om det er en valid query før du bare smækker den på

File: src/UICrafter.Core/Components/Render/RenderTextBox.razor [hunk 6]
```diff
@@ -27,9 +29,15 @@
     {
         if (string.IsNullOrEmpty(Textbox.Textbox.JsonField))
         {
-            Content = "No JsonField specified.";
```
Comment (Aagaard00) [inline src/UICrafter.Core/Components/Render/RenderTextBox.razor:30]: Burden den bare at default til reponse hvis der ikke er skrevet noget?

**Comment by Aagaard00** [comment]
Skulle det være muligt at disable "clean-up"? Hvis man nu gerne vil have printet i JSON format

File: src/UICrafter.Client/Components/Properties/Components/TextboxProperties.razor (hunk 4 above)
Comment (Nephim) [reply src/UICrafter.Client/Components/Properties/Components/TextboxProperties.razor]: What do you want to change? Doing this RadioChanged is cleaner as the values needs to be switched?

File: src/UICrafter.Core/Components/Render/RenderAPIButton.razor [hunk 7]
```diff
@@ -86,7 +86,7 @@
 
             NotifyUser(Color.Error, $"{(int)response.StatusCode} - {response.StatusCode}", Severity.Error);
 
-            var jsonResponse = JsonSerializer.Serialize(new { StatusCode = $"{(int)response.StatusCode} - {response.ReasonPhrase}", Data = responseData });
+            string jsonResponse = "{ StatusCode =" + response.StatusCode + "-" + response.ReasonPhrase + ", Data = responseData }";
```
Comment (Nephim) [inline src/UICrafter.Core/Components/Render/RenderAPIButton.razor]: This is not valid json? what does this do?

File: src/UICrafter.Core/Components/Render/RenderTextBox.razor [hunk 8]
```diff
@@ -1,3 +1,5 @@
+@using DevLab.JmesPath
+@using Newtonsoft.Json.Linq
```
Comment (Nephim) [inline src/UICrafter.Core/Components/Render/RenderTextBox.razor]: Newtonsoft illegal!

**Review by Nephim — CHANGES_REQUESTED** [review]
A few things, maybe we need to go over it. idk

File: src/UICrafter.Core/Components/Render/RenderTextBox.razor (hunk 8 above)
Comment (Gormiee) [reply src/UICrafter.Core/Components/Render/RenderTextBox.razor]: JmesPath.NET depends on Newtonsoft

**Pull request merged**